functions/parse.js

```diff
@@ -16,6 +16,9 @@ const parse = (version, options) => {
   try {
     return new SemVer(version, options)
   } catch (er) {
+    if (throwErrors) {
```

Comment:
I wonder if the `return null`'s above should also be errors that are thrown? Currently even with `throwErrors` as `true` there are still cases where `null` could come back

Reply:
Very good point and in trying to fix it I am reminded of discussions from #541.  Perhaps the very tiny performance hit for bad semver shouldn't outweigh the benefits of having a single consistent place for semver to parse versions.

Reply:
Hmm I was thinking keep the if statements but I’v move them into the try and have them throw an error instead of return null

If the performance overhead is the exception maybe it’s fine it’s just the error path?

Reply:
Oh I see the checks already exist so having them here was already redundant 

https://github.com/npm/node-semver/blob/503a4e52fe2b1c6ed1400d33149f7733c8361eed/classes/semver.js#L18-L26